install_devel.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# Installation Script for Local Development
+
+# Install Git submodules
+git submodule update --init
+
+# Install RPM packages
+sudo zypper install git php7 php7-fileinfo php7-gettext php7-json php7-mbstring \
+    php7-opcache php7-sqlite php-composer nodejs8 npm8 ImageMagick
+
+# Install NodeJS packages
+sudo npm install -g gulp-cli
+cd skins/Chameleon
+npm install
+cd ../..
+
+# Download MediaWiki 1.27.4
+wget https://releases.wikimedia.org/mediawiki/1.27/mediawiki-1.27.4.tar.gz
+tar -xvzf mediawiki-1.27.4.tar.gz
+cp -rf mediawiki-1.27.4/* .
+rm -r mediawiki-1.27.4 mediawiki-1.27.4.tar.gz
+
+# Download MediaWiki extensions
+
+function download {
+    wget https://extdist.wmflabs.org/dist/extensions/$1
+    tar -xvzf $1
+    rm $1
+}
+
+cd extensions
+download AbuseFilter-REL1_27-2072d2f.tar.gz
+download Auth_remoteuser-REL1_27-89faa0c.tar.gz
+download CategoryTree-REL1_27-b454f2c.tar.gz
+download intersection-REL1_27-38cdaf6.tar.gz
+download MultiBoilerplate-REL1_27-bb13f76.tar.gz
+download ReplaceText-REL1_27-7676bf8.tar.gz
+download RSS-REL1_27-d945221.tar.gz
+download UserMerge-REL1_27-31ea86d.tar.gz
+download UserPageEditProtection-REL1_27-8affdda.tar.gz
+cd ..
```

Comment:
There's an easier way which is also closer to the production setup:

Add the `openSUSE:infrastructure:wiki` repo,  then `zypper in mediawiki_1_27-openSUSE` - and finally do some symlink magic to symlink everything that is not in this git repo ;-)

Reply:
Now I switched to openSUSE's mediawiki packages and linked to git directory. Here are two small issues:

1. Need package build for openSUSE Tumbleweed. The content of package might be the same, but different system version may worry users.
2. Package hhvm is missing so the dependency is broken. However, local development doesn't really need it.

Reply:
Tumbleweed builds are enabled now.

For hhvm - this requirement is probably caused by `#!/usr/bin/hhvm -f` in maintenance/hiphop/run-server, and shouldn't cause any real-world problems.